static/app/components/searchQueryBuilder/tokens/freeText.tsx

```diff
@@ -491,16 +516,27 @@ function SearchQueryBuilderInputInternal({
               value,
             }),
             shouldCommitQuery: false,
+            replaceRawSearchKey: !suppressBlur.current && !ctrlABeingPressed.current,
           });
+
           resetInputValue();
         }}
         onCustomValueCommitted={value => {
-          // if we haven't changed anything, just search
-          if (value.trim() === trimmedTokenValue) {
+          // If we haven't changed anything, just search. Note, we override this if
+          // the search query builder has raw search replacement enabled, as we want to
+          // always replace any free text tokens with raw search keys.
+          if (value.trim() === trimmedTokenValue && !shouldReplaceRawSearchKey) {
```

Comment:
Are you sure you want to do this? If the text hasn't changed it seems like we can continue with the search regardless of the replaceRawSearchKey prop.

For the case where the input is empty, you would still want to call `handleSearch` if they hit enter